terraform-provider-constellation/internal/provider/attestation_data_source.go

```diff
@@ -130,7 +169,12 @@ func (d *AttestationDataSource) Read(ctx context.Context, req datasource.ReadReq
 		resp.Diagnostics.AddError("Converting SNP attestation", err.Error())
 	}
 	verifyFetcher := measurements.NewVerifyFetcher(sigstore.NewCosignVerifier, d.rekor, d.client)
-	fetchedMeasurements, err := verifyFetcher.FetchAndVerifyMeasurements(ctx, data.ImageVersion.ValueString(),
+
+	imageVersion := data.ImageVersion.ValueString()
+	if imageVersion == "" {
+		imageVersion = d.version // Use provider version as default.
```

Comment:
We should add a log message here, saying that it falls back to the default version.